inmoose/pycombat/covariates.py

```diff
@@ -150,6 +158,98 @@ def __init__(self, message):
         super().__init__(self.message)
 
 
+def format_covar_mod(covar_mod, batch, cov_missing_value=None):
+    """Format the covariate table in model matrix.
+
+    Arguments
+    ---------
+    covar_mod : list or matrix
+        model matrix (dataframe, list or numpy array) contaning one or multiple covariates
+    batch : array or list or :obj:`inmoose.utils.factor.Factor`
+        batch indices
+    cov_missing_value : str
+        Option to choose the way to handle missing covariates
+        `None` raise an error if missing covariates and stop the code
+        "remove" remove samples with missing covariates and raise a warning
+        "fill" handle missing covariates, by creating a distinct covariate per batch
+        (default: "raise")
+
+    Returns
+    -------
+    remove_sample : list
+        sample indices to remove from the analysis
+    covar_mod : dataframe
+        model matrix (dataframe) contaning covariates in interger format
+    """
+    if (type(covar_mod) == list) | (type(covar_mod) == np.ndarray):
+        if (type(covar_mod[0]) == list) | (type(covar_mod[0]) == np.ndarray):
+            covar_mod = pd.DataFrame(covar_mod).transpose()
+        else:
+            covar_mod = pd.DataFrame(covar_mod)
+        covar_mod.columns = ["col_" + str(col_nb) for col_nb in list(covar_mod.columns)]
+
+    # check for float type with decimal (excluding nan) to identify potential continue_variable
+    continue_variable = [
+        col
+        for col in covar_mod.columns
+        if True
+        in [
+            (ele % 1) > 0
+            for ele in covar_mod[col].dropna().to_list()
+            if type(ele) == float
```

Comment:
No it will not, I don't know how to check for this kind of variable, but even if it's not identify as continious here, it will stop the execution after because the age will become a confounding variable

Reply:
Ok I see, maybe just warn the user that a covariate is of type int, if it's by purpose then he won't change, if it's not he will take a look?

Reply:
Done, I also add 2 tests for the error and warning message